l2geth/rollup/sync_service_test.go

```diff
@@ -132,11 +132,7 @@ func TestSyncServiceTransactionEnqueued(t *testing.T) {
 		},
 	})
 
-	// Run an iteration of the eloop
-	err = service.sequence()
-	if err != nil {
-		t.Fatal("sequencing failed", err)
-	}
+	go service.syncQueueToTip()
```

Comment:
Why was this changed? IMO it is super error prone to use go-routines which we don't know how many iterations they'll run for inside tests. Better to explicitly run a single iteration (or N iterations for N transactions) inside.

Reply:
`syncQueueToTip` does less things than `sequence` and calling it once is a single iteration. This starts a goroutine that runs until `syncQueueToTip` returns. This was added because if it does not run in a goroutine, it blocks due to the addition of `txCh` and waiting explicitly for transactions to be added to the blockchain after processing them